src/Microsoft.OData.Core/UriParser/Resolver/NormalizedModelElementsCache.cs

```diff
@@ -89,26 +93,76 @@ public List<IEdmTerm> FindTerms(string qualifiedName)
             return null;
         }
 
+        /// <summary>
+        /// Find all navigation sources that match the <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The case-insensitive name to match.</param>
+        /// <returns>A list of matching navigation sources, or null if no navigation source matches the name.</returns>
+        public List<IEdmNavigationSource> FindNavigationSources(string name)
+        {
+            if (navigationSourcesCache.TryGetValue(name, out List<IEdmNavigationSource> results))
+            {
+                return results;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Find all operation imports that match the <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The case-insensitive name to match.</param>
+        /// <returns>A list of matching operation imports, or null if no operation import matches the name.</returns>
+        public List<IEdmOperationImport> FindOperationImports(string name)
+        {
+            if (operationImportsCache.TryGetValue(name, out List<IEdmOperationImport> results))

```

Comment:
same question here..can we have several operations with the same name in a model or what's the rationale of outing a `List<IEdmOperationImport>`

Reply:
In theory, we can have multiple similar names that only differ by case (e.g. `FooBar` and `fooBar`). For operations, we can also have overloads with the same name but different parameters.

Reply:
so here do you check for the params too?